commands/import.go

```diff
@@ -102,40 +106,51 @@ func checkFilesExist(files []string) error {
 	return nil
 }
 
-// setRolling determines what the current rolling and chunk settings should be
-// based on the values already in the database and the values supplied on the
-// command line by the user.
-func setRolling(dbExists bool, dbIsRolling bool, dbCurrChunk int, dbTotalChunks int,
-	userIsRolling bool, userCurrChunk int, userTotalChunks int, cfgDefaultChunks int) (config.RollingStaticCfg, error) {
+// parseFlags validates the user supplied flags against the current state of
+// the target database and determines what settings should be set in the
+// rolling configuration
+func parseFlags(dbExists bool, dbIsRolling bool, dbCurrChunk int, dbTotalChunks int,
+	userIsRolling bool, userCurrChunk int, userTotalChunks int, cfgDefaultChunks int,
+	deleteOldData bool) (config.RollingStaticCfg, error) {
 
 	cfg := config.RollingStaticCfg{}
 
-	// Can uncomment this check if we want to preserve old behavior with regards to non-rolling databases
-	// if dbExists && !dbIsRolling {
-	// 	return cfg, errors.New("\n\t[!] New data cannot be imported into an existing non-rolling database")
-	// }
-
 	// a user-provided value for either of the chunk options implies rolling
 	if userTotalChunks != -1 || userCurrChunk != -1 {
 		userIsRolling = true
 	}
 
+	// ensure the user specifies a rolling import if the database exists
+	// and is not a rolling database.
+	if !deleteOldData && (dbExists && !dbIsRolling) && !userIsRolling {
+		return cfg, errors.New(
+			"\t[!] New data cannot be imported into a non-rolling database. " +
+				"Run with --rolling to convert this database into a rolling database.",
+		)
+	}
+
+	// set cfg.TotalChunks
 	if userTotalChunks != -1 { // user gave the total number of chunks via command line
 		// it's currently an error to try to reduce the total number of chunks in an existing rolling database
-		if dbExists && dbIsRolling && userTotalChunks < dbTotalChunks  {
+		if dbExists && dbIsRolling && userTotalChunks < dbTotalChunks {
 			return cfg, fmt.Errorf(
-				"\n\t[!] Cannot modify the total number of chunks in an existing database [ %d ]",
+				"\t[!] Cannot modify the total number of chunks in an existing database [ %d ]",
 				dbTotalChunks,
 			)
 		}
 
 		// use the user-provided value
 		cfg.TotalChunks = userTotalChunks
 	} else { // user didn't specify the total number of chunks
+
 		if !dbExists && !userIsRolling {
 			// if the database doesn't exist and wasn't specified to be rolling
 			// then assume only one chunk
 			cfg.TotalChunks = 1
+		} else if deleteOldData && dbExists && !dbIsRolling && !userIsRolling {
```

Comment:
Missed this in the old version. I'm not sure whether total chunks and current chunk matter if cfg.Rolling isn't set, but we might as well have a coherent config even if the values aren't currently used.